File: models/docente.py
from sqlalchemy import Column, Integer, String, Boolean
from config.db import Base

class Docente(Base):
    __tablename__ = "docente"

    id_docente = Column(Integer, primary_key=True, index=True)
    nombres = Column(String(100), nullable=False)
    apellidos = Column(String(100), nullable=False)
    cedula = Column(String(20), unique=True, nullable=False)
    correo = Column(String(120), unique=True, nullable=False)
    contraseña = Column(String(255), nullable=False)
    especialidad = Column(String(100))
    estado = Column(Boolean, default=True)

File: src/models/Docente.js
export class Docente {
  constructor({
    nombres,
    apellidos,
    cedula,
    correo,
    especialidad,
    telefono,
    estado,
    id_docente,
  }) {
    this.nombres = nombres;
    this.apellidos = apellidos;
    this.cedula = cedula;
    this.correo = correo;
    this.especialidad = especialidad;
    this.telefono = telefono || null;
    this.estado = estado;
    this.id_docente = id_docente;
  }
}
